lib/types.ts

```diff
@@ -63,6 +63,9 @@ export type NonNever<T extends {}> = Pick<T, { [K in keyof T]: T[K] extends neve
 /** Merge 2 types, properties types from the latter override the ones defined on the former type */
 export type Merge<M, N> = Omit<M, Extract<keyof M, keyof N>> & N;
 
+/** Mark some properties as required, leaving others unchanged */
+export type MarkRequired<T, RK extends keyof T> = Exclude<T, RK> & Required<Pick<T, RK>>;
```

Comment:
NOTE: i think the other way to implement is to use `Merge` type defined above. `Merge<T, Required<Pick<T, RK>>>` just as a side note

Reply:
I've actually used `Merge` here in the beginning, but then the created types grow overcomplicated (not really readable when you hover over them in vscode). Because this merges disjoint parts of the same type, we don't have to explicitly setup overrides, and the output is a bit cleaner.